WcaOnRails/lib/custom_i18n_scanner.rb

```diff
@@ -0,0 +1,76 @@
+require 'i18n/tasks/scanners/file_scanner'
+class WCAFileScanner < I18n::Tasks::Scanners::FileScanner
+  include I18n::Tasks::Scanners::RelativeKeys
+  include I18n::Tasks::Scanners::OccurrenceFromPosition
+
+  def active_record?(model)
+    # FIXME: Find a way to determine this dynamically ?
+    model != "contact"
+  end
+
+  def extract_model_attr(key)
+    model = key.gsub(/^(\w+)\..*$/, '\1').singularize
+    # Some dirty tricks to assign the detected field to the "correct" model
+    if model == "devise"
+      model = "user"
+    elsif model == "admin"
+      model = "person"
+    elsif model == "oauth"
+      model = "doorkeeper/application"
+    end
```

Comment:
Oh you can use switch for strings...

About the admin->person for sure I can explain, here is what the custom scanner report when you don't do the substitution:
>+--------+-----------------------------------------+------------------------------------------+
| Locale | Key                                     | Value in other locales or source         |
+--------+-----------------------------------------+------------------------------------------+
|  all   | activerecord.attributes.admin.countryId | app/views/admin/edit_person.html.erb:27  |
|  all   | activerecord.attributes.admin.dob       | app/views/admin/edit_person.html.erb:27  |
|  all   | activerecord.attributes.admin.gender    | app/views/admin/edit_person.html.erb:27  |
|  all   | activerecord.attributes.admin.name      | app/views/admin/edit_person.html.erb:27  |
|  all   | activerecord.attributes.admin.wca_id    | app/views/admin/edit_person.html.erb:27  |

The thing is that for the majority of controllers the model used has the same name, so I can assume that a form input will use the corresponding activerecord attribute; but the `admin` controller modifies `person` objects, so I need to change the model name so that the scanner reports the use of the attribute on the correct model.